### PR TITLE
SADAR Changes, added 'Unguided' HE/WP rocket types, Reworked AP.

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Divides SADAR Rockets into two sections, Laser-Guided and Unguided.

Dumbfire rockets have a higher payload but do not detonate on the tile you hit on, have far lower direct hit damage, and they keep traveling until they hit anything.

Laser-Guided rockets have a lower payload, and keep the current SADAR behavior of detonating on the tile you have clicked on.

AP Rocket has been reworked into dealing 360 damage and 200 penetration, it no longer sunders. It now has .5 falloff and is called a "Kinetic penetrator", being based off of LOSAT missiles. (325/150/100)->(360/200/0)

EDIT: Removed aim mode, no longer doubled amounts of rockets, renamed Dumbfire rockets to unguided after feedback.
SADAR Rockets in general now just have 14 tiles of range. 

## Why It's Good For The Game
Makes the SADAR into a true bunker buster, with AP rockets being your xeno removal wonder but having absolutely no sunder if the xenomorph doesn't die, which makes it less punishing to get hit and survive.

Aim mode makes the Miniscopeless SADAR much more useable, and allows for more dynamic use of it with the newly added dumbfire rockets.

Unguided rockets should allow for more unique use of the SADAR itself and busting defenses with it, while Laser-Guided rockets have old style SADAR of precision application of hurt rather than total devastation.

AP now works like the high-end hurt machine it should be, but it no longer punishes with insanely high sunder, use LG HE for that.

## Changelog
:cl:
balance: SADAR Rework- Has Aim mode (1.4 seconds to aim.) Unguided rockets that go past your clicks added, they have a much higher payload compared to normal rockets, AP rocket reworked from 325Dam/150Pen/100Sunder to 360Dam/200Pen/0Sunder.
balance: SADAR rockets in general have 14 tiles of max travel distance in general, includes unguided and laser guided missiles.
/:cl:

### CODE REVIEW

**Comment by novaepee** [comment]
>SADAR
>aim mode
>what

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -1670,13 +1670,21 @@ datum/ammo/bullet/revolver/tp44
 /datum/ammo/rocket/do_at_max_range(turf/T, obj/projectile/P)
 	drop_nade(T.density ? P.loc : T)
 
+/datum/ammo/rocket/dumbfire
+	damage = 50
+	flags_ammo_behavior = AMMO_ROCKET|AMMO_SUNDERING // We want this one to specifically go over onscreen range.
+
+/datum/ammo/rocket/drop_nade(turf/T)
+	explosion(T, 0, 7, 7, 2)
```
Comment (Lumipharon) [inline code/modules/projectiles/ammo_datums.dm:1678]:
```suggestion
	explosion(T, 0, 7, 0, 2)
```
Makes it clearer what its actual effect is. Also fucking huge heavy explosion.

File: code/modules/projectiles/magazines/specialist.dm [hunk 2]
```diff
@@ -128,26 +128,38 @@
...
-	name = "\improper 84mm white-phosphorus rocket"
+	name = "\improper 84mm 'L-G' white-phosphorus rocket"
 	icon_state = "wp_rocket"
 	default_ammo = /datum/ammo/rocket/wp
-	desc = "A highly destructive warhead that bursts into deadly flames on impact. Use this in hand to deconstruct it."
+	desc = "A highly destructive warhead that bursts into deadly flames on impact. Due to being laser-guided, it will hit exactly where you aim, however the payload is smaller due to the internal space required for this. Use this in hand to deconstruct it."
+
+/obj/item/ammo_magazine/rocket/sadar/wp/dumb
+	name = "\improper 84mm 'Dumbfire' white-phosphorus rocket"
+	icon_state = "wp_rocket"
```
Comment (Lumipharon) [inline code/modules/projectiles/magazines/specialist.dm:159]:
```suggestion
```
It already inherits this, no need to specify it gain

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -1778,6 +1786,11 @@ datum/ammo/bullet/revolver/tp44
 	max_range = 30
 	sundering = 100
 
+/datum/ammo/rocket/wp/dumbfire
+	flags_ammo_behavior = AMMO_ROCKET|AMMO_INCENDIARY|AMMO_SUNDERING
+	damage = 50
+	effect_radius = 7
```
Comment (Lumipharon) [inline code/modules/projectiles/ammo_datums.dm:1792]: I can smell the salt already.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by lKiseki** [comment]
This is one heck of a powercreep PR if I've ever seen one.
Let's go down the list, I suppose.

## General changes
- You gave SADAR aim mode. I'm not going to elaborate on this one, and all I will ask is: _Why?_
- You doubled the amount of rockets gained in requisitions orders. except for AP, which is instead 1 point cheaper. Very much unnecessary, and hardly justified, given what I'm looking at.

## AP rocket
Does no sunder now. Alright, cool... _But,_ here's the catch, which makes it even worse than before: **320 damage increased to 360**, and **150 penetration raised to 200**, as if the previous value wasn't already enough to skip armor entirely.
0.5 damage falloff is practically meaningless -- after 6 tiles traveled, that's 3 less damage. It would have to travel 80 tiles to go down to the previous value of 320, which was already very high by xeno health standards.
So with that in mind, let's take a look at the implications of these changes:
- Instantly kills Runners and Wraiths, **skipping crit.** Yes, this does so much damage that they just die.
- Will knock the following castes **straight into crit**: Drones, Sentinels, Defenders, Hunters, Bulls, Warriors, Hivelords, Carriers, and Boilers.
- The following castes **barely survive**: Widow, Praetorian (they have EXACTLY 360 HP at ancient, so this puts them at 0), Defiler, Crusher, Gorger, Queen, Shrike, and King.

You're not gonna be caring much about the sunder when you've been instantly knocked into crit, are you?
Oh, right. I'd like to mention this: this was tested using Ancient xeno health values.
May God have mercy on you if you run into this before Ancient.

## Dumbfire rockets
New rocket type that, instead of detonating on the tile you target and/or being limited in range to your screen, has 20 range and will detonate when it hits something. Has 50 damage on a direct hit, which is pretty bad. The catch? **Ignites on a direct hit, 7 tiles of heavy and light explosion.** High friendly fire potential.
AND, as if that weren't enough? **It deals 100 sunder.** I double checked this on local. I'm hoping it's an unintended side effect of giving it `AMMO_SUNDERING` and not replacing the `sundering = 100` from its parent type.
If it isn't, this just makes it into a way better HE rocket. Full sunder, incendiary, _and_ a big heavy explosion, all wrapped into one neat package, just for you.

## Concluding thoughts
**Massive powercreep**. I have no better words to describe it.
AP rockets have pointless changes that only make it much more worse to deal with it.
Dumbfire rockets are just better HE rockets.
Aim mode (_why?_)

Needless to say, I heavily disapprove of this PR as a whole.

**Comment by Oscari-pog** [comment]
Straight up just a massive buff worded in a way that tries to make it seem like it has downsides

**Comment by iLikeMemes** [comment]
I smell a marine main.

**Comment by MereWalker** [comment]
Not a fan of these changes. To make sadar used again miniscope would be the way to go. That isn't the world we live in anymore. May as well scrape the pr alone with the sadar.

**Comment by Naaanii** [comment]
Let's go.

**Comment by TiviPlus** [comment]
Aimmode hard no
AP rocket undecided
dumb rocket yes
doubling rocket count no

**Comment by Araulius** [comment]
Just PB the xenos with your rocket bro. It's not hard.  AP has no splash.

**Comment by Losenis** [comment]
>Instantly kills Runners and Wraiths

Speedmerge, fuck those fucking fuckers

**Comment by Al-1ce** [comment]
You have to pick between wanting these to be a common and free thing from roundstart intended to be a finisher, or something rare and expensive that can turn a rouny into red paste in an instant. Both are valid options. Minding the aim mode, this PR actually seems to fix the problem of SADARs being more or less useless.

**Comment by Al-1ce** [comment]
SADAR's have existed in a sort of grey-zone for a long time now. It seems we can't really decide weather or not we want these to be powerful and punishing, or weak and easy to shrug off. You can't find an even middle ground for a rocket launcher. It has to be one extreme or the other. Completely useful, albeit rare and expensive -- or situationally useless and free from roundstart.

Honestly? I'd rather this than the current state of things. SADARs should be costly and rare at the expensive of being much more threatening. 

With all of that said, aim-mode on an RPG is silly and isn't present in grenade launchers for a reason.

**Comment by Al-1ce** [comment]
SADAR's have existed in a sort of grey-zone for a long time now. It seems we can't really decide weather or not we want these to be powerful and punishing, or weak and easy to shrug off. You can't find an even middle ground for a rocket launcher. It has to be one extreme or the other. Completely useful, albeit rare and expensive -- or situationally useless and free from roundstart.

Honestly? I'd rather this than the current state of things.

**Comment by Naaanii** [comment]
Make each shell 20 points too to compensate.

File: code/modules/projectiles/ammo_datums.dm (hunk 3 above)
Comment (BraveMole) [reply code/modules/projectiles/ammo_datums.dm:1792]: Wow no

File: code/modules/projectiles/ammo_datums.dm [hunk 4]
```diff
@@ -1778,6 +1786,11 @@ datum/ammo/bullet/revolver/tp44
 	max_range = 30
 	sundering = 100
 
+/datum/ammo/rocket/wp/dumbfire
```
Comment (BraveMole) [inline code/modules/projectiles/ammo_datums.dm:1789]: what the hell is that name

File: code/modules/projectiles/magazines/specialist.dm [hunk 5]
```diff
@@ -128,26 +128,37 @@
...
+	name = "\improper 84mm 'L-G' high-explosive rocket"
+	desc = "A warhead for the RL-152 rocket launcher. Carries a bogstandard HE warhead that explodes. Due to being laser-guided, it will hit exactly where you aim, however the payload is smaller due to the internal space required for this.  When empty, use this frame to deconstruct it."
 	caliber = CALIBER_84MM
 	icon_state = "rocket"
 	w_class = WEIGHT_CLASS_NORMAL
 	max_rounds = 1
 	default_ammo = /datum/ammo/rocket
 	reload_delay = 60
 
+/obj/item/ammo_magazine/rocket/sadar/dumb
```
Comment (BraveMole) [inline code/modules/projectiles/magazines/specialist.dm:140]: Find another name than dumb. Unguided ?

File: code/modules/reqs/supplypacks.dm [hunk 6]
```diff
@@ -363,6 +369,12 @@ WEAPONS
 	cost = 5
 	available_against_xeno_only = TRUE
 
+/datum/supply_packs/weapons/rpg_wp_dumb
```
Comment (BraveMole) [inline code/modules/reqs/supplypacks.dm:372]: no

File: code/modules/reqs/supplypacks.dm [hunk 7]
```diff
@@ -347,14 +347,20 @@ WEAPONS
 
 /datum/supply_packs/weapons/rpg_regular
 	name = "RL-152 SADAR HE rocket"
-	contains = list(/obj/item/ammo_magazine/rocket/sadar)
+	contains = list(/obj/item/ammo_magazine/rocket/sadar, /obj/item/ammo_magazine/rocket/sadar)
+	cost = 6
+	available_against_xeno_only = TRUE
+
+/datum/supply_packs/weapons/rpg_regular_dumb
```
Comment (BraveMole) [inline code/modules/reqs/supplypacks.dm:354]: no doubling

File: code/modules/projectiles/ammo_datums.dm [hunk 8]
```diff
@@ -1670,13 +1670,21 @@ datum/ammo/bullet/revolver/tp44
 /datum/ammo/rocket/do_at_max_range(turf/T, obj/projectile/P)
 	drop_nade(T.density ? P.loc : T)
 
+/datum/ammo/rocket/dumbfire
+	damage = 50
+	flags_ammo_behavior = AMMO_ROCKET|AMMO_SUNDERING // We want this one to specifically go over onscreen range.
+
+/datum/ammo/rocket/drop_nade(turf/T)
+	explosion(T, 0, 7, 0, 2)
```
Comment (BraveMole) [inline code/modules/projectiles/ammo_datums.dm:1678]: You are mixing datums order here. do all parent rocket stuff then define children

File: code/modules/projectiles/ammo_datums.dm [hunk 9]
```diff
@@ -1670,13 +1670,21 @@ datum/ammo/bullet/revolver/tp44
 /datum/ammo/rocket/do_at_max_range(turf/T, obj/projectile/P)
 	drop_nade(T.density ? P.loc : T)
 
+/datum/ammo/rocket/dumbfire
+	damage = 50
```
Comment (BraveMole) [inline code/modules/projectiles/ammo_datums.dm:1674]: 50 damage, instead of 200 ...?

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by Al-1ce** [comment]
> Make each shell 20 points too to compensate.

The rocket launcher itself is currently free, so this is a good compromise. Maybe somewhere around 10-15 will be better for the long-term once we see how this works through testing.

**Comment by Al-1ce** [comment]
> The SADAR is free? When did that happened.

T-160s can be found in the 'Specialized' tab of the marine equipment vendor in limited quantities. Always have been.

**Comment by Lumipharon** [comment]
> > The SADAR is free? When did that happened.
> 
> T-160s can be found in the 'Specialized' tab of the marine equipment vendor in limited quantities. Always have been.

that ain't a sadar.

**Comment by Naaanii** [comment]
> > Make each shell 20 points too to compensate.
> 
> The rocket launcher itself is currently free, so this is a good compromise. Maybe somewhere around 10-15 will be better for the long-term once we see how this works through testing.

It's not. Recoiless Rifle is free which is the T-160.
SADAR is the big brother of the RR that's 120 points.

**Comment by comfyorange** [comment]
#### Test Merge Deployed

##### Server Instance
TGMC

##### Merged By
naaanii

##### Revision
Origin: af73cb5a187ee4479e42fb50efc36e8fbcfdfc30
Pull Request: 74857997a38efc10d283630bd7832f99e37ca697
Server: a82a7b9a2e34a0152938b45c435209b017c35eac

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Tiviplus

##### Revision
Origin: af73cb5a187ee4479e42fb50efc36e8fbcfdfc30
Pull Request: 74857997a38efc10d283630bd7832f99e37ca697
Server: 2592483a876e876dcb90a7aa0c54b0fa595184c4

##### Comment

**Comment by zzurge** [comment]
id like to say that i literally almost never see the sadar deployed...like at all anymore, people dont buy it,its even more expensive than the gamer shotgun

**Comment by Starfirejordan** [comment]
Test AP, insta crits majority of xenos and i mean criting near death not entered crit, and gibs almost any xeno that isnt past mature
just.. put AP back to how it was, dont make AP a butt fuck rocket

**Comment by Goblinu** [comment]
> id like to say that i literally almost never see the sadar deployed...like at all anymore, people dont buy it,its even more expensive than the gamer shotgun

As someone who frequently see's SADARS deployed, being on the revieving end of one is the least enjoyable thing in the game. To be instacritted by a single shot is just fun when youre a t3

**Comment by Araulius** [comment]
throwing this out there. Consider a rocket type that does medium damage, but giga slowdown.

**Comment by Al-1ce** [comment]
>SADAR/whatever

Guh. Not sure why I was thinking of the T-160. My previous points still stand.

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: af73cb5a187ee4479e42fb50efc36e8fbcfdfc30
Pull Request: 74857997a38efc10d283630bd7832f99e37ca697
Server: 99de39021678299d1ff1c63ce1c4f70ed66a98d9

##### Comment

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Tiviplus

##### Revision
Origin: 1a0d30d9b48151b309e3ce11f7ab98481c77c9a9
Pull Request: 74857997a38efc10d283630bd7832f99e37ca697
Server: 7adad29020eeb4be86efb0ddff82c4d7eb2491b2

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: ba11e493e0e54387094b1aee3f723c962e51dc58
Pull Request: 74857997a38efc10d283630bd7832f99e37ca697
Server: 1c1e4b98bed5b49a08101b5d08554feba05f2f0b

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: de3c2c8ebf8e37758d277c25b7907d8059064f00
Pull Request: 74857997a38efc10d283630bd7832f99e37ca697
Server: 54f6b1801c5e375491da3123d1bedf3c5c653b16

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 74ddc18f3425d2a813310b95d0a2e0f96909bce9
Pull Request: 74857997a38efc10d283630bd7832f99e37ca697
Server: 0630093ff9a7e29366983fccc1d30b0735259a83

##### Comment

**Comment by comfyorange** [comment]
#### Test Merge Removed

**Pull request closed without merging**